### PR TITLE
Increase appliance demand in CHN

### PR DESCRIPTION
**Pull request by robinhasse**
- change delimiter of config files from `,` to `;`
  - consistent with REMIND configs
  - more convenient to handle in Excel as one should not need to change the application delimiter settings all the time
  - no need for quotation of cells in the config (regional parameterisation requires `,` that cannot be confused with the file delimiter anylonger
- Major bugfix: regional switch values
  - Regional scenario assumptions did not work as expected before. The old implementation did not control the region column which lead to random assignments of values to regions.
  - Also, assigning values to native EDGE-B regions instead of region groups had no effect due to a bug.
  - Both issues should be fixed now. This changes the default results (esp. for space heating) where econCoef were assigned in a wrong way.
- We increase the projections of UE demand for appliances and lighting in CHN in consideration of current trends. This increases electricity demand substantially in this region.

**. . .**      old version
**---** after the bug fix (all regions are affected, not just CHN)
**___** adjustment of appliances in CHN

grafik
grafik

### CODE REVIEW

File: R/readConfig.R [hunk 1]
```diff
@@ -103,7 +102,7 @@ readConfig <- function(config = getSystemFile("config", "configTest.csv", packag
             lstVal[reg]
         } else if (reg %in% regions) {
           # individual regions
-          df[reg, "value"] <- lstVal[reg]
+          df[reg == df$region, "value"] <- lstVal[reg]
```
Comment (ricardarosemann) [inline R/readConfig.R:105]: I don't fully understand the reason for this change. Or is it to make sure that the code does not stumble over non-existing regions? And I thought the lintr-safe formulation would then be ```df[reg == df[["region"]], "value"]?

**Review by ricardarosemann — APPROVED** [review]
Thank you! Very important and helpful, and I especially like your refactoring of ```buildScenInput```.
I only found a funny typo and I have two very minor comments.

File: R/buildScenInput.R [hunk 2]
```diff
@@ -258,12 +105,19 @@ buildScenInput <- function(config,
     }
 
     return(valueMap)
-  }
+  } else if (subtype == "scen_assump_corr") {
+    ## not sure yet if necessary ====
+
+    # TODO: is it?
```
Comment (hagento) [inline R/buildScenInput.R:111]: This should not be needed anymore. We currently give the regression parameters from the output of `mredgebuildings::calcEfficiencyRegression` as the target `feueEffPars`. These have already been corrected, making this step unnecessary.

**Review by hagento — APPROVED** [review]
Thanks for the nice work! I really like the refactoring of `buildScenInput` as well, which looks way cleaner now. Other than the comment on the `subtype = scen_assump_corr` in the aforementioned function, I have nothing to add to Ricarda's suggestions.

File: R/buildScenInput.R [hunk 3]
```diff
@@ -287,5 +141,111 @@ buildScenInput <- function(config,
...
+}
+
+
+#' Convert config to data frame with values for each region
+#'
+#' @param config data.frame with one line per scenario and switches as columns
+#' @param regions character vector of all regions
+#' @param switches character vector of switch names (without prefix)
+#' @param prefix common switch prefix that will be removed
+#' @returns data.frame witch switch values for each scenario and region
```
[the remark this answers is not in the document]
Comment (robinhasse) [reply R/buildScenInput.R:217]: I feel I could need them, though :)

File: R/buildScenInput.R [hunk 4]
```diff
@@ -91,157 +52,43 @@ buildScenInput <- function(config,
...
-            mutate(variable = var,
-                   scenario = scen)
-        }
-      )
-    )
+    ## temporal convergence assumptions ====
 
-    scenAssumpSpeed <- tmpReg %>%
-      pivot_wider(names_from = "variable", values_from = "value") %>%
+    .expandToRegions(config, regions, c("speed", "speed_fullconv")) %>%
```
[the remark this answers is not in the document]
Comment (robinhasse) [reply R/buildScenInput.R:71]: I agree. Thanks.

File: R/buildScenInput.R (hunk 2 above)
Comment (robinhasse) [reply R/buildScenInput.R:111]: Okay, cool. I removed the respective code.

**Pull request merged**